packages/net/_test.pony

```diff
@@ -669,3 +673,124 @@ class _TestTCPProxyNotify is TCPConnectionNotify
 
   fun ref connect_failed(conn: TCPConnection ref) =>
     _h.fail_action("sender connect failed")
+
+class _TestTCPConnectionFailed is UnitTest
+  fun name(): String => "net/TCPConnectionFailed"
+
+  fun ref apply(h: TestHelper) =>
+    h.expect_action("connection failed")
+
+    let host = "127.0.0.1"
+    let port = "7669"
+
+    try
+      let connection = TCPConnection(
+        h.env.root as AmbientAuth,
+        object iso is TCPConnectionNotify
+          let _h: TestHelper = h
+
+          fun ref connected(conn: TCPConnection ref) =>
+            _h.log("log: connected to a non-existent host and port")
+            _h.fail("fail: connected to a non-existent host and port")
+            _h.complete(false)
+            conn.close()
+
+          fun ref connect_failed(conn: TCPConnection ref) =>
+            _h.log("log: connection failed")
```

Comment:
given the complete action, is there any need for this log?

Reply:
Will remove